core/encoding_jbig2.go

```diff
@@ -262,24 +313,30 @@ func newJBIG2DecoderFromStream(streamObj *PdfObjectStream, decodeParams *PdfObje
 			}
 		}
 	}
-
-	if decodeParams != nil {
-		if globals := decodeParams.Get("JBIG2Globals"); globals != nil {
-			var err error
-
-			globalsStream, ok := globals.(*PdfObjectStream)
-			if !ok {
-				err = errors.Error(processName, "jbig2.Globals stream should be an Object Stream")
-				common.Log.Debug("ERROR: %s", err.Error())
-				return nil, err
-			}
-			encoder.Globals, err = jbig2.DecodeGlobals(globalsStream.Stream)
-			if err != nil {
-				err = errors.Wrap(err, processName, "corrupted jbig2 encoded data")
-				common.Log.Debug("ERROR: %s", err)
-				return nil, err
-			}
-		}
+	// if no decode params provided end fast.
+	if decodeParams == nil {
+		return encoder, nil
+	}
+	// set image parameters.
+	encoder.UpdateParams(decodeParams)
+	// check if the
```

Comment:
Please add the rest of the comment :). Or remove it if it is not necessary.

Reply:
No need to comment that.:)